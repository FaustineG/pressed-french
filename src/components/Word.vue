<script setup lang="ts">
defineProps<{ word: string; visible: boolean; found: boolean }>()
</script>

<template>
  <div class="word">
    <div v-for="letter in word.split('')" :key="letter" class="letter" :class="{ 'not-found': !found  && visible}">
      {{ visible ? letter : '-' }}
    </div>
  </div>
</template>

<style scoped lang="scss">
.word {
  display: flex;
  min-width: 10rem;
  .letter {
    display: inline-flex;
    background: white;
    border: 1px solid var(--background-color);
    width: 2rem;
    height: 2rem;
    font-size: 2rem;
    justify-content: center;
    align-items: center;
    color: black
  }

  .not-found{
    color: red;
  }

  
}
</style>
