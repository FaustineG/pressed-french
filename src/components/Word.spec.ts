import { describe, expect, it } from 'vitest'
import Word from './Word.vue'
import { mount } from '@vue/test-utils'

const initComponent = (propsData = {}) => {
  return mount(Word, {
    props: {
      word: 'patate',
      visible: true,
      found: false,
      ...propsData
    }
  })
}

describe('Word', () => {
  it('should match snapshot', () => {
    const wrapper = initComponent()
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('should display letters', () => {
    const wrapper = initComponent()
    const allLettersElements = wrapper.findAll('.letter')
    expect(allLettersElements).toHaveLength(6)

    expect(allLettersElements[0].text()).toEqual('p')
    expect(allLettersElements[1].text()).toEqual('a')
    expect(allLettersElements[2].text()).toEqual('t')
    expect(allLettersElements[3].text()).toEqual('a')
    expect(allLettersElements[4].text()).toEqual('t')
    expect(allLettersElements[5].text()).toEqual('e')
  })
  it('should display only hyphens when not visible', () => {
    const wrapper = initComponent({ visible: false })
    expect(wrapper.findAll('.letter')).toHaveLength(6)
    wrapper.findAll('.letter').forEach((letterElement) => {
      expect(letterElement.text()).toEqual('-')
    })
  })
})
