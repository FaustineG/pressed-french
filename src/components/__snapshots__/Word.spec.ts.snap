// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Word > should match snapshot 1`] = `
"<div data-v-88bbdb11="" class="word">
  <div data-v-88bbdb11="" class="letter not-found">p</div>
  <div data-v-88bbdb11="" class="letter not-found">a</div>
  <div data-v-88bbdb11="" class="letter not-found">t</div>
  <div data-v-88bbdb11="" class="letter not-found">a</div>
  <div data-v-88bbdb11="" class="letter not-found">t</div>
  <div data-v-88bbdb11="" class="letter not-found">e</div>
</div>"
`;
